### PR TITLE
removed autoformat when no necessary

### PR DESCRIPTION
**Pull request by joanvr**
## What's changed?
Remove the call to maybeAutoFormat on some places

## What's your motivation?
Autoformat in some places it's not strictly needed, due to the produced changes, and because the limited scope of the autoFormat it can lead to unwanted results.

## Anything in particular you'd like reviewers to focus on?
I've added couple of tests to check for extra scenarios, but maybe I missed some corner case where the autoFormat was needed

## Anyone you would like to review specifically?
@timtebeek @rpau 

## Have you considered any alternatives or workarounds?
Maybe trying to fix the AutoFormat (more specifically the TabsAndIndentsVisitor) could be a better fix, but it seems kinda tricky and complex thing to do, with much bigger (and hard to test) impact. 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases

### CODE REVIEW

File: src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/NoRequestMappingAnnotationTest.java [hunk 1]
```diff
@@ -281,7 +281,7 @@ public ResponseEntity<List<String>> getUsersPost(String userId) {
               
               @RestController
               public class UsersController {
-                  @PostMapping(value = "/user/{userId}/edit", produces = {MediaType.APPLICATION_JSON_VALUE})
+                  @PostMapping(value = "/user/{userId}/edit", produces = { MediaType.APPLICATION_JSON_VALUE })
```
Comment (joanvr) [inline src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/NoRequestMappingAnnotationTest.java:284]: This one in particular, for me, was a "side-effect" of applying always auto-format and doing more changed than needed. The original parameter was in this format, but in the result, we are removing the spaces without reason. I thing it's better to preserve original format in this recipe.

File: src/main/java/org/openrewrite/java/spring/NoRequestMappingAnnotation.java [hunk 2]
```diff
@@ -93,7 +94,8 @@ public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ct
                 // Remove the argument
                 if (requestMethodArg.isPresent() && methodArgumentHasSingleType(requestMethodArg.get()) && resolvedRequestMappingAnnotationClassName != null) {
                     if (a.getArguments() != null) {
-                        a = maybeAutoFormat(a, a.withArguments(ListUtils.map(a.getArguments(), arg -> requestMethodArg.get().equals(arg) ? null : arg)), ctx);
+                        //a = maybeAutoFormat(a, a.withArguments(ListUtils.map(a.getArguments(), arg -> requestMethodArg.get().equals(arg) ? null : arg)), ctx);
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/spring/NoRequestMappingAnnotation.java:97]:
```suggestion
```

File: src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/NoRequestMappingAnnotationTest.java (hunk 1 above)
Comment (timtebeek) [reply src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/NoRequestMappingAnnotationTest.java:284]: Agreed

**Review by timtebeek — APPROVED** [review]
